### ISSUE TITLE
Overlapping

### ISSUE THREAD

**Issue by massimomannoni** [jamesmh/coravelpro#44]
Hi James, using coravelpro in a cluster env I verified an unexpected behaviors. Starting on both node coravel pro interfaces the overlapping is available for different job types but not for the opposite case. The result is the possibility to have 2 jobs running on the same time avoiding in fact the prevent overlapping function.

**Comment by jamesmh** [comment]
Yes that is correct. Right now there's no distributed lock that can "tell" each node when a given job is running. The lock is in-memory per node/process instance.

This is a future feature but not in place yet 😢

**Comment by siganberg** [comment]
I was doing the same testing to evaluate whether the Pro version can be used for our requirements and unfortunately I found similar behavior. I have 2 pods running on Kubernetes load balance and they both trigger. And sometimes if I delete the schedule, one pod still executing or if I create the schedule only one executes it even if I allow overlap.  

I was already thinking of using the regular version and just building a client (job executor) and server api (job manager) instead of combining the functionality in one service but it will be a lot of work.

**Comment by jamesmh** [comment]
There's an approach taken by the open-source library around distributed locking on jobs -> [https://github.com/jamesmh/coravel#does-coravel-support-distributed-locking](https://github.com/jamesmh/coravel#does-coravel-support-distributed-locking).

The same type of approach would work with Coravel Pro.
